src/build.d

```diff
@@ -0,0 +1,1131 @@
+    enforce(build.among("release", "debug"), "BUILD must be 'debug' or 'release'");
+
+    // detect Model
+    auto model = env.getDefault("MODEL", detectModel);
+    env["MODEL_FLAG"] = "-m" ~ env["MODEL"];
+
+    // detect PIC
+    version(Posix)
+    {
+        // default to PIC on x86_64, use PIC=1/0 to en-/disable PIC.
+        // Note that shared libraries and C files are always compiled with PIC.
+        bool pic;
+        version(X86_64)
+            pic = true;
+        else version(X86)
+            pic = false;
+        if (environment.get("PIC", "0") == "1")
+            pic = true;
+
+        env["PIC_FLAG"]  = pic ? "-fPIC" : "";
+    }
+    else
+    {
+        env["PIC_FLAG"] = "";
+    }
+
+    env.getDefault("GIT_HOME", "https://github.com/dlang");
+    env.getDefault("SYSCONFDIR", "/etc");
+    env.getDefault("TMP", "/tmp");
+    env.getDefault("PGO_DIR", srcDir.buildPath("pgo"));
+    auto d = env.getDefault("D", "dmd");
+    env.getDefault("C", d.buildPath("backend"));
+    env.getDefault("TK", d.buildPath("tk"));
+    env.getDefault("ROOT", d.buildPath("root"));
+    env.getDefault("EX", d.buildPath("examples"));
+    auto generated = env.getDefault("GENERATED", srcDir.dirName.buildPath("generated"));
+    auto g = env.getDefault("G", generated.buildPath(os, build, model));
+    mkdirRecurse(g);
+
+    env.getDefault("HOST_CXX", "c++");
+    env.getDefault("AR", "ar");
+    env.getDefault("GIT", "git");
+    auto cxxVersion = execute([env["HOST_CXX"], "--version"]).output;
+    env.getDefault("CXX_KIND", !cxxVersion.find("gcc", "Free Software")[0].empty ? "g++" : "clang++");
+
+    env.getDefault("HOST_DMD", "dmd");
+
+    // Auto-bootstrapping of a specific host compiler
+    if (env.getDefault("AUTO_BOOTSTRAP", "0") != "0")
+    {
+        auto hostDMDVer = "2.074.1";
+        writefln("Using Bootstrap compiler: %s", hostDMDVer);
+        auto hostDMDRoot = env["G"].buildPath("host_dmd-"~hostDMDVer);
+        auto hostDMDBase = hostDMDVer~"."~os;
+        auto hostDMDURL = "http://downloads.dlang.org/releases/2.x/"~hostDMDVer~"/dmd."~hostDMDBase;
+        env["HOST_DMD"] = hostDMDRoot.buildPath("dmd2", os, os == "osx" ? "bin" : "bin"~model, "dmd");
+        env["HOST_DMD_PATH"] = env["HOST_DMD"];
+        // TODO: use dmd.conf from the host too (in case there's a global or user-level dmd.conf)
+        env["HOST_DMD_RUN"] = env["HOST_DMD"];
+        if (!env["HOST_DMD"].exists)
+        {
+            writefln("Downloading DMD %s", hostDMDVer);
+            auto curlFlags = "-fsSL --retry 5 --retry-max-time 120 --connect-timeout 5 --speed-time 30 --speed-limit 1024";
+            hostDMDRoot.mkdirRecurse;
+            ("curl " ~ curlFlags ~ " " ~ hostDMDURL~".tar.xz | tar -C "~hostDMDRoot~" -Jxf - || rm -rf "~hostDMDRoot).spawnShell.wait;
+        }
+    }
+    else
+    {
+        env["HOST_DMD_PATH"] = ["which", env["HOST_DMD"]].execute.output.strip;
+        env["HOST_DMD_RUN"] = env["HOST_DMD"];
+    }
```

Comment:
Auto-bootstrapping doesn't make so much sense anymore as it did with the Makefile (after all a D compiler is already required to run this script).
However, the use case of looking the compilation to a specific D version still exists.